internal/appsec/waf/waf.go

```diff
@@ -624,6 +751,18 @@ func decNbLiveCObjects() {
 	atomic.AddUint64(&nbLiveCObjects, ^uint64(0))
 }
 
+// gostring returns the Go version of the C string `str`, copying at most `len` bytes from the original string.
+func gostring(str *C.char, len C.uint64_t) (string, error) {
+	if str == nil {
+		return "", ErrInvalidArgument
+	}
+	goLen := C.int(len)
+	if C.uint64_t(goLen) != len {
+		return "", ErrInvalidArgument
```

Comment:
Do you know any bitwise operation trick to safely convert it to `int` when it's safe to do so?

Can we do something like:

```go
if C.int(len) <= 0 { /* overflow? */ }
```

Also, it is worth having some unit tests for this condition.

Reply:
Seems like this comment is a duplicate, resolving